Startup Kit Activity

Gr3gg0r opened the pull request:
### Description

This PR contains a Life feature implementation. This feature was done for the benefit of SK Activity. The PR includes the backend, frontend, and integration between these two.

### Tested On

- [x]  Web

Gr3gg0r committed:
feat(life): add new life collection.

Solve part 2 of the sk-activity.

### Fix file extension
One of the migration files is mistakenly using the `.tsx` extension, but since no XML-like code is involved, it should be renamed to `.ts`.

deguzman-carlo reviewed (commented):
Good first commit 🚀

Gr3gg0r committed:
feat(part-3): Implement graphql on life .

Implement:
- add resolver for Life type
- add schema
- Handle mutation and query

deguzman-carlo commented on schema/Mutation.graphql:
```diff
@@ -84,4 +84,16 @@ type Mutation {
+    Create a new life data.
+    """
+    createLife(
+        firstName: String!
+        lastName: String!
+        description: String!
+        birthday: DateTime!
+        hobbies: [String!]!
+        title: String!
```
Might be good if we put this in an `input` type instead

Gr3gg0r replied:
Fix on new update.

deguzman-carlo commented on schema/Query.graphql:
```diff
@@ -28,4 +28,14 @@ type Query {
+    """
+    Fetch list of the lives.
+    """
+    listLives: [Life]
```
suggest that we can return this as `[Life!]!` so we don't need to handle null/undefined values.
We either care about wether it is an empty array, and the life data it contains

deguzman-carlo commented on src/server/schema/resolvers/mutations/createLife.ts:
```diff
@@ -0,0 +1,27 @@
+        title,
+    });
+
+    const life = await collections.lives.findOne({ _id: operation.insertedId });
```
we can directly return this query instead

deguzman-carlo commented on src/server/schema/resolvers/queries/getLife.ts:
```diff
@@ -0,0 +1,19 @@
+    const transformedLives = {
+        ...life,
+        id: life._id,
+        fullName: `${life.firstName} ${life.lastName}`, // Dynamically create fullName
```
We will not do it here, rather, we will do it in `Life` resolver

deguzman-carlo replied:
same goes for the `._id`, you can refer to `User` resolver regarding implementation

Gr3gg0r replied:
Fix on newly commit

deguzman-carlo commented on src/server/schema/resolvers/types/Life.ts:
```diff
@@ -0,0 +1,8 @@
+import { GraphQLLifeResolvers } from '../definitions';
+
+const LifeGraphQL: GraphQLLifeResolvers = {
+    fullName: root => `${root.firstName} ${root.lastName}`,
+    id: root => root._id,
```
oh you already did it here, no need to do inside the query

Gr3gg0r replied:
yeah forgot about this one hahaha.

Gr3gg0r committed:
feat(part-4): Integrate life to frontend

The integration phase for the new life feature has been implemented.

The representation was added for Life View and Life List page.

Gr3gg0r committed:
fix(review): Resolve comment on PR

Multiple changes,
- Change to using graphql input.
- Return some query directly

deguzman-carlo commented on src/app/pages/public/LifePage/LifeViewPage/index.tsx:
```diff
@@ -0,0 +1,93 @@
+            <Result
+                extra={
+                    <Button onClick={goBack} type="primary">
+                        Back Home
```
For static text,  you may want to use translation for this with `react-i18next`

deguzman-carlo commented on src/app/pages/public/LifePage/index.tsx:
```diff
@@ -0,0 +1,90 @@
+        );
+    }
+
+    const columns = [
+        {
+            title: 'No',
+            dataIndex: 'key',
+            key: 'key',
+        },
+        {
+            title: 'Full Name',
+            dataIndex: 'fullName',
+            key: 'fullName',
+        },
+        {
+            title: 'Birthday',
+            dataIndex: 'birthday',
+            key: 'birthday',
+            render: text => new Date(text).toLocaleDateString(),
+        },
+    ];
+
+    // Data mapping
+    const dataSource = data?.listLives.map((item, index) => ({
+        key: index + 1,
+        id: item.id,
+        fullName: item.fullName,
+        birthday: item.birthday,
+        description: item.description,
+        firstName: item.firstName,
+        lastName: item.lastName,
+        hobbies: item.hobbies,
```
You can memoized these values instead using `useMemo`
